docs/extend/custom_dynamic_components.md

```diff
@@ -0,0 +1,129 @@
+# Custom dynamic components
+
+To create dynamic behavior, you got to choose: Either you wrap a bunch of `core components` and/or `custom components` inside an `dynamic core component`, or you go the whole way and create a `custom dynamic component` with a corresponding Vue.js counterpart.
+
+## Async wrapper around static components
+
+Create a `custom static component` in `app/matestack/components/static/component.rb`
+
+```ruby
+class Components::Static::Component < Matestack::Ui::StaticComponent
+
+  def response
+    components {
+      div id: "my-component" do
+        plain "I'm a static component!"
+        plain DateTime.now.strftime('%Q')
+      end
+    }
+  end
+
+end
+```
+
+and add it to the Example Page, wrapping it into an *async component* to make it *dynamic*! The *async component* is a `core component` and therefore does not need the *custom_* prefix.
+
+Now, the page will respond with static content, but our component rerenders (visible by looking at the timestamp) whenever *\"my_event\"* happens. This event may be triggered by all kinds of other components, for example an `onclick` component, as shown below:
+
+```ruby
+class Pages::ExamplePage < Matestack::Ui::Page
+
+  def response
+    components {
+      div id: "div-on-page" do
+        onclick emit "my_event"
+
+        async rerender_on: "my_event" do
+          custom_some_component
```

Comment:
does not match the component name `Components::Static::Component` --> custom_static_component